src/wmtk/multimesh/utils/extract_child_mesh_from_tag.cpp

```diff
@@ -0,0 +1,121 @@
+#include "extract_child_mesh_from_tag.hpp"
+#include <map>
+#include <wmtk/EdgeMesh.hpp>
+#include <wmtk/Mesh.hpp>
+#include <wmtk/MultiMeshManager.hpp>
+#include <wmtk/PointMesh.hpp>
+#include <wmtk/Primitive.hpp>
+#include <wmtk/TetMesh.hpp>
+#include <wmtk/TriMesh.hpp>
+
+
+namespace wmtk::multimesh::utils {
+
+std::shared_ptr<Mesh> extract_and_register_child_mesh_from_tag(
+    Mesh& m,
+    const std::string& tag,
+    const long tag_value,
+    const PrimitiveType& pt)
+{
+    assert(m.top_simplex_type() >= pt);
+    auto tag_handle = m.get_attribute_handle<long>(tag, pt);
+    return extract_and_register_child_mesh_from_tag_handle(m, tag_handle, tag_value);
+}
+
+
+std::shared_ptr<Mesh> extract_and_register_child_mesh_from_tag_handle(
+    Mesh& m,
+    const MeshAttributeHandle<long>& tag_handle,
+    const long tag_value)
+{
+    auto tags = m.create_const_accessor(tag_handle);
+    PrimitiveType pt = tag_handle.primitive_type();
+
+    std::vector<Tuple> tagged_tuples;
+    for (auto t : m.get_all(pt)) {
+        if (tags.const_scalar_attribute(t) == tag_value) {
+            tagged_tuples.emplace_back(t);
+        }
+    }
+
+    switch (pt) {
+    case PrimitiveType::Vertex: throw("not implemented");
```

Comment:
You can add a simple test case for that using `CHECK_THROWS()`. This test case can then later be modified when there is something actually happening.